enterprise/internal/insights/compression/observability.go

```diff
@@ -0,0 +1,46 @@
+package compression
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+
+	"github.com/sourcegraph/sourcegraph/internal/metrics"
+	"github.com/sourcegraph/sourcegraph/internal/observation"
+)
+
+type operations struct {
+	worker       *observation.Operation
+	getCommits   *observation.Operation
+	countCommits *prometheus.CounterVec
+}
+
+func newOperations(observationContext *observation.Context) *operations {
+	worker := observationContext.Operation(observation.Op{
+		Name: "CommitIndexer.Run",
+		Metrics: metrics.NewOperationMetrics(
+			observationContext.Registerer,
+			"insights_commit_indexer",
+			metrics.WithCountHelp("Total number of commit indexer executions"),
+		),
+	})
+
+	getCommits := observationContext.Operation(observation.Op{
+		Name: "CommitIndexer.GetCommits",
+		Metrics: metrics.NewOperationMetrics(
+			observationContext.Registerer,
+			"insights_commit_indexer_fetch",
+			metrics.WithCountHelp("Time for the commit indexer to fetch commits from gitserver."),
+		),
+	})
+
+	countCommits := promauto.NewCounterVec(prometheus.CounterOpts{
+		Name: "insights_commit_indexer_commits_added",
+		Help: "Total number of commits added to the commit index for this repo",
```

Comment:
```suggestion
		Help: "Number of commits added to the commit index",
```

This won't be tracking totals, just counts over some interval of time. Likewise since we aren't going to break out per-repo cardinality, this is an aggregated value.